### PR TITLE
feat: adjust buildchart to be able to render foreground bars

### PR DESCRIPTION
**Pull request by ikondrat**
## What has changed and why?

We need to be able to render foreground bar for BuildChart to show entire dataset comparing to current selection defined by filters.
Screenshot 2026-07-28 at 08 56 19

## How has it been tested?

Covered by unit tests. No braking changes

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

* **New Features**
  * Bar charts now show a grey “full total” background context when a filter is active, alongside the filtered bars.
  * Updated bar highlighting to better reflect selected, non-selected, and non-selectable states.
  * Tooltips now use a shared formatter to present total vs in-filter context with accurate percentage calculations.
  * Chart grid top padding is now configurable via `gridTopPx`.

* **Tests**
  * Expanded test coverage for filtering/selection rendering, tooltip formatting behavior (including HTML escaping and edge cases), and `gridTopPx`.

### CODE REVIEW

File: lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts [hunk 1]
```diff
@@ -68,30 +84,87 @@ export function buildEchartsOption(
...
+            hasActiveFilter && item.filteredCount !== undefined ? item.filteredCount : item.count;
+        const isDimmed = hasAnySelected && !item.selected;
+        return {
+            value: foregroundCount,
+            itemStyle: {
+                color: isDimmed ? BAR_COLOR_DIMMED : BAR_COLOR,
+                opacity: item.selectable === false ? 0.45 : 1
+            }
+        };
+    });
```
[the remark this answers is not in the document]
Comment (ikondrat) [reply lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts:101]: we should able to select several bars as well

File: lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts [hunk 2]
```diff
@@ -68,30 +79,70 @@ export function buildEchartsOption(
...
+        : null;
+
+    const series = hasActiveFilter ? [backgroundSeries, foregroundSeries] : [foregroundSeries];
+
     return {
         backgroundColor: 'transparent',
         tooltip: {
             trigger: 'axis',
             axisPointer: { type: 'shadow' },
-            appendTo: 'body',
```
Comment (mihnea-necsulescu) [inline lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts:76]: why was this removed?

File: lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts [hunk 3]
```diff
@@ -25,6 +24,8 @@ interface BuildEchartsOptionOptions {
     totalCount?: number;
     /** Bar orientation (default 'vertical'). */
     orientation?: BarChartOrientation;
+    /** Top chart-grid padding in px (default 16). */
+    gridTopPx?: number;
```
Comment (mihnea-necsulescu) [inline lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts:28]: is this going to be used in the follow up prs? it's currently only used in tests

File: lightly_studio_view/src/lib/components/BarChart/buildTooltipFormatter/buildTooltipFormatter.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            const filteredValue = params[1].value; // foreground series = filtered count
+            if (totalValue !== filteredValue) {
+                const totalPct =
+                    totalCount > 0 ? ` (${formatPercent(totalValue / totalCount)})` : '';
+                const filteredPct =
+                    totalCount > 0 ? ` (${formatPercent(filteredValue / totalCount)})` : '';
+                return `<b>${escape(params[0].name)}</b><br/>Total: <b>${totalValue}</b>${totalPct}<br/>In filter: <b>${filteredValue}</b>${filteredPct}`;
+            }
+        }
+        const [{ name, value }] = params;
```
Comment (mihnea-necsulescu) [inline lightly_studio_view/src/lib/components/BarChart/buildTooltipFormatter/buildTooltipFormatter.ts:21]: can the params be an empty array? should this case be handled?

File: lightly_studio_view/src/lib/components/BarChart/buildTooltipFormatter/buildTooltipFormatter.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import escape from 'lodash-es/escape';
```
Comment (mihnea-necsulescu) [inline lightly_studio_view/src/lib/components/BarChart/buildTooltipFormatter/buildTooltipFormatter.ts:1]:
```suggestion
import { escape } from 'lodash-es/escape';
```

File: lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts (hunk 3 above)
Comment (ikondrat) [reply lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts:28]: yes, it will be used in BarChart

File: lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts (hunk 2 above)
Comment (ikondrat) [reply lightly_studio_view/src/lib/components/BarChart/buildEchartsOption.ts:76]: merge problem, will fix it

**Review by mihnea-necsulescu — APPROVED** [review]
